common/json.go

```diff
@@ -76,16 +71,24 @@ func (j *JSONHTTPClient) Patch(ctx context.Context,
 		return nil, err
 	}
 
-	// empty response body should not be parsed as JSON since it will cause ajson to err
-	if len(body) == 0 {
-		return nil, nil //nolint:nilnil
+	return parseJSONResponse(res, body)
+}
+
+func (j *JSONHTTPClient) Delete(ctx context.Context, url string, headers ...Header) (*JSONHTTPResponse, error) {
+	res, body, err := j.HTTPClient.Delete(ctx, url, addAcceptJSONHeader(headers)) //nolint:bodyclose
+	if err != nil {
+		return nil, err
 	}
 
 	return parseJSONResponse(res, body)
 }
 
 // parseJSONResponse parses the given HTTP response and returns a JSONHTTPResponse.
 func parseJSONResponse(res *http.Response, body []byte) (*JSONHTTPResponse, error) {
+	// empty response body should not be parsed as JSON since it will cause ajson to err
+	if len(body) == 0 {
+		return nil, nil //nolint:nilnil
+	}
```

Comment:
it looks like every CRUD operation does this check (except GET, could be a bug), pulled it to this method since all of them call it at the end.